keras_nlp/models/whisper/whisper_audio_to_speech_lm.py

```diff
@@ -0,0 +1,498 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Whisper Seq2Seq LM (Language Model)."""
+
+import copy
+import os
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.generative_task import GenerativeTask
+from keras_nlp.models.whisper.whisper_audio_to_speech_lm_preprocessor import (
+    WhisperAudioToSpeechLMPreprocessor,
+)
+from keras_nlp.models.whisper.whisper_backbone import WhisperBackbone
+from keras_nlp.models.whisper.whisper_presets import backbone_presets
+from keras_nlp.utils.keras_utils import is_xla_compatible
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioToSpeechLM")
+class WhisperAudioToSpeechLM(GenerativeTask):
```

Comment:
I think we should still probably call this seq2seq for consistency, after all, we didn't call it `BartTextToTextLM`.